src/Rule/AtLeast.php

```diff
@@ -5,70 +5,51 @@
 namespace Yiisoft\Validator\Rule;
 
 use Attribute;
-use Yiisoft\Validator\FormatterInterface;
-use Yiisoft\Validator\Result;
-use Yiisoft\Validator\Rule;
-use Yiisoft\Validator\ValidationContext;
+use Closure;
+use Yiisoft\Validator\Rule\Trait\HandlerClassNameTrait;
+use Yiisoft\Validator\Rule\Trait\RuleNameTrait;
+use Yiisoft\Validator\RuleInterface;
 
 /**
- * Checks if at least {@see AtLeast::$min} of many object attributes are filled.
+ * Checks if at least {@see AtLeast::$min} of many attributes are filled.
  */
 #[Attribute(Attribute::TARGET_PROPERTY)]
-final class AtLeast extends Rule
+final class AtLeast implements RuleInterface
 {
+    use HandlerClassNameTrait;
+    use RuleNameTrait;
+
     public function __construct(
         /**
-         * @var string[] The list of required attributes that will be checked.
+         * The list of required attributes that will be checked.
          */
-        private array $attributes,
+        public array $attributes,
         /**
-         * @var int The minimum required quantity of filled attributes to pass the validation.
+         * The minimum required quantity of filled attributes to pass the validation.
          * Defaults to 1.
          */
-        private int $min = 1,
+        public int $min = 1,
```

Comment:
Why parameters of rule is public?
I think rule must be immutable.

Reply:
It can be done in another PR

Reply:
But this changed in this PR. In master all parameters is private.

Reply:
And now they are public because of two different scopes